pennylane/interfaces/autograd.py

```diff
@@ -411,18 +411,24 @@ def _get_jac_with_caching():
             return cached_jac["jacobian"]
 
         jacs = []
-        for t in tapes:
-            if isinstance(device, qml.devices.experimental.Device):  # pragma:  no-cover
-                # cant test until we integrate device with shot vector
-                shot_vector = t.shots.shot_vector if t.shots.has_partitioned_shots else None
-            else:
-                shot_vector = device.shot_vector
-            g_tapes, fn = gradient_fn(t, shots=shot_vector, **gradient_kwargs)
+        if isinstance(device, qml.devices.experimental.Device):  # pragma:  no-cover
+            # cant test until we integrate device with shot vector
+            shot_vector = (
+                tapes[0].shots.shot_vector if tapes[0].shots.has_partitioned_shots else None

```

Comment:
We will be transitioning so that the gradient transforms can accept shots from the tape.  Therefore they will be able to check individual tapes, and no longer need the `shots` keyword argument.